src/field/binary/extension.rs

```diff
@@ -0,0 +1,305 @@
+//! contains implementation of binary extension fields using tower field arithmetic in multilinear basis as defined in [DP23b](https://eprint.iacr.org/2023/1784.pdf)
+use std::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign},
+};
+
+use super::BinaryField;
+use crate::field::FiniteField;
+
+/// Binary extension field GF_{2^{2^K}} using binary towers arithmetic as explained in Section 2.3 of [DP23b](https://eprint.iacr.org/2023/1784.pdf)
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BinaryFieldExtension<const K: usize>
+where [(); 1 << K]: {
+  /// coefficients of field element
+  pub coefficients: [BinaryField; 1 << K],
+}
```

Comment:
This guy is a const expr fiend now!!

Reply:
everywhere I go, I see compile time constants now :)